homeassistant/components/device_tracker/tomato.py

```diff
@@ -39,16 +45,19 @@ class TomatoDeviceScanner(DeviceScanner):
     def __init__(self, config):
         """Initialize the scanner."""
         host, http_id = config[CONF_HOST], config[CONF_HTTP_ID]
+        port = config[CONF_PORT]
         username, password = config[CONF_USERNAME], config[CONF_PASSWORD]
+        self.ssl, self.verify_ssl = config[CONF_SSL], config[CONF_VERIFY_SSL]
 
         self.req = requests.Request(
-            'POST', 'http://{}/update.cgi'.format(host),
+            'POST', 'http{}://{}:{}/update.cgi'.format(
+                "s" if self.ssl else "", host, port
```

Comment:
If ssl is true, port still defaults to 80. Is that what you were expecting?

Reply:
Oh, I hadn't even really thought much about that to be honest.  I'll throw a small update in for that so that it'll default to 80 or 443 depending on if there is/isn't ssl enabled.